### DISCUSSION TITLE
How to do certificate authentication

### DISCUSSION THREAD

**Discussion by kirankg15** [rajanadar/VaultSharp#299] [Q&A]
I would like to implement cert authentication with hashicorp vault.

I have gone through vaultsharp library but I could not find the methods related to generationOfCert , WritingcertstoVault,
AuthenticateWithCerts.

Could you please help me on this where I can find the supporting methods in vaultsharp library

**Comment by rajanadar** [comment]
hi @kirankg15 

Can you please elaborate your use case? Do you want to just consume vault secrets using a certificate you have OR do you want VaultSharp to actually generate the certs etc.?

If it is the first, then VaultSharp has the feature.

Please let me know.
